src/core/vector/qgsvectorlayerutils.cpp

```diff
@@ -431,9 +431,20 @@ bool QgsVectorLayerUtils::validateAttribute( const QgsVectorLayer *layer, const
 
     if ( !exempt )
     {
-      valid = valid && !value.isNull();
 
-      if ( value.isNull() )
+      const bool allowMulti { field.editorWidgetSetup().config().value( QStringLiteral( "AllowMulti" ), false ).toBool() };
```

Comment:
Can we move this into a `bool QgsFieldFormatter::validate()` (or similar) method?

Reply:
maybe, with what kind of signature? The formatter has no knowledge of constraints.

Reply:
`isNull()`?